lib/drivers/weitai/weiTaiUSB.js

```diff
@@ -0,0 +1,825 @@
+      postbolus = postbolus.done();
+      postbolus.deviceId = 'MTM-I';
+      bolusArray.push(postbolus);
+      return bolusArray;
+    },
+
+    buildValue(originValue){
+      let value = this.formatDecimal(originValue,2);
+      let res =((value*1000)/25);
+      // console.log(value);
+      let floorRes = Math.floor(res);
+      let floor = floorRes*25;
+      if(res > floorRes) {
+        floor = (floor + 25)/1000;
+      }else{
+        floor = floor/1000;
+      }
+      return floor;
+    },
+    formatDecimal(originnum, decimal) {
+      let num = originnum.toString();
+      let index = num.indexOf('.');
+      let resNum = '';
+      if (index !== -1) {
+        resNum = num.substring(0, decimal + index + 1);
+      } else {
+        resNum = num.substring(0);
+      }
+      return parseFloat(resNum).toFixed(decimal);
+    },
+    fetchData(progress, data, cb) {
+      // debug('in fetchData', data);
+      let records = [];
+      let incomingA = data.incomingA;
+      const inComeRes = WeitaiUSB.parsePacket(incomingA,cfg,cb);  //inComeRes:{BloodGlucoses:[],BasalRates:[],BolusRates:[]}
+      data.BloodGlucoses = inComeRes.BloodGlucoses;
+      data.BasalRates = inComeRes.BasalRates;
+      data.BolusRates = inComeRes.BolusRates;
+      data.records = records;
+      return cb(null, data);
+    },
+
+    processData(progress, data, cb) {
+      progress(100);
+      let bloodRes = this.buildBlood(data.BloodGlucoses);
+      let basalRes = this.buildBasal(data.BasalRates);
+      // let basalRes = this.buildBasal(testBasal);
+      let bolusRes = this.buildBolus(data.BolusRates);
+      // let bolusRes = this.buildBolus(testArray);
+      basalRes = basalRes.length > 1 ? basalRes:[];
+      let post_records = [].concat(bloodRes,basalRes,bolusRes);
+      if(!post_records.length){
+        return cb({code:'E_WEITAI_NO_UPLOAD'}, data);
+      }
+      data.post_records = post_records;
+      return cb(null, data);
+    },
+
+    uploadData(progress, data, cb) {
+      progress(0);
+      const sessionInfo = {
+        delta: cfg.delta,
+        deviceTags: cfg.deviceInfo.tags,
+        // deviceManufacturers: cfg.deviceInfo.manufacturers,
+        // deviceModel: cfg.deviceInfo.model,
+        deviceManufacturers: ['Weitai'],
+        deviceModel: 'weitaiupload',
+        deviceSerialNumber: cfg.deviceInfo.serialNumber || 'sdsdgdfdg',
+        deviceId: cfg.deviceInfo.deviceId || 'MTM-I',
+        start: sundial.utcDateString(),
+        timeProcessing: cfg.tzoUtil.type,
+        tzName: cfg.timezone,
+        version: cfg.version,
+      };
+      // return false;
+      cfg.api.upload.toPlatform(
+        data.post_records, sessionInfo, progress, cfg.groupId,
+        (err, result) => {
+          progress(100);
+
+          if (err) {
+            debug(err);
+            debug(result);
+            cb(err, data);
+          }
+          data.cleanup = true;
+          return cb(null, data);
+        }, 'jellyfish',
```

Comment:
```suggestion
        }, 'dataservices',
```
This should be changed to 'dataservices`, as `jellyfish` has been deprecated. `dataservices` will perform proper validation to make sure the data is correct.